### PR TITLE
Model app manager deployer and provider

### PR DESCRIPTION
**Pull request by facuspagnuolo**
**MERGE AFTER #60** (change base branch to the [refactor branch](https://github.com/zeppelinos/zos-cli/tree/refactor_branch) before merging)

Currently we have JS contract wrappers carrying out a lot of responsibilities. I propose splitting them into smaller models in favor of moving them to `zos-lib` later. There is a common pattern between all these wrappers:
- deploying a new instance of the contract being wrapped
- wrapping an already deployed instance
- carrying out wrapped contract's domain behavior

Then, my proposal is to have three different models here:
- a deployer: being in charge of deploying a contract and all its dependencies returning a JS wrapper for that contract
- a provider: being in charge of building a JS wrapper fetching whatever is needed from the blockchain
- a JS wrapper: being in charge of carrying out contract's behavior (e.g. AppManager wrapper should know how to create a proxy) 

This PR starts with the `AppManager`. I needed to do some improvements to the `Stdlib` model too. It was to coupled with the `AppManager` and it was doing a lot of things like deploying, providing, installing dependencies, etc.

### CODE REVIEW

File: src/zos-lib/stdlib/StdlibInstaller.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  constructor(stdlibNameAndVersion) {
+    this.stdlibNameAndVersion = stdlibNameAndVersion
+  }
+
+  async call() {
+    await this.installDepIfTestEnv()
+    return new Stdlib(this.stdlibNameAndVersion)
+  }
+
+  async installDepIfTestEnv() {
```
Comment (spalladino) [inline src/zos-lib/stdlib/StdlibInstaller.js]: This method should be named IfNOTTestEnv

File: src/zos-lib/app_manager/AppManagerProvider.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import AppManagerWrapper from "./AppManagerWrapper"
+import ContractsProvider from "../../models/ContractsProvider"
+
+export default class AppManagerProvider {
```
[the remark this answers is not in the document]
Comment (facuspagnuolo) [reply src/zos-lib/app_manager/AppManagerProvider.js]: Let's use plain objects for now

**Comment by facuspagnuolo** [comment]
agree with your comments @spalladino, please review

**Comment by spalladino** [comment]
I was thinking about exporting by default the `from` function from the Providers, rather than an object that includes it, but this also works. Merging!

**Pull request merged**